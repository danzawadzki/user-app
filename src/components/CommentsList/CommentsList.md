Default CommentsList component with all props:

```jsx
const user = {
	author: {
		name: 'Random User',
		avatar: {
			small: 'https://randomuser.me/api/portraits/thumb/men/65.jpg',
			regular: 'https://randomuser.me/api/portraits/med/men/65.jpg',
			large: 'https://randomuser.me/api/portraits/men/65.jpg'
		}
	},
	comment:
		'Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula.',
	timestamp: '10/10/2018'
};
<CommentsList comments={[user, user, user, user]} />;
```

CommentsList component with out props:

```jsx
<CommentsList />
```
