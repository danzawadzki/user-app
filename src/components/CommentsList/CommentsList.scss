@import '../../assets/styles/colors';

.CommentsList {
	list-style: none;
	padding: 0;
	margin: 0;

	&__item {
		padding: 2.25rem 0;
		border-bottom: 1px solid $app__borders;
	}
}
