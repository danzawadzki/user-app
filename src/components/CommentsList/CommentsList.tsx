import * as React from 'react';
import './CommentsList.scss';
import Comment, { IComment } from '../Comment/Comment';

export interface IComments {
	/** Array of objects representing comments */
	comments?: IComment[];
}

/**
 * Comments list component.
 *
 * @version 1.0.0
 */
const CommentsList = ({ comments = [] }: IComments) => {
	if (!comments || comments.length === 0) {
		return <p className="CommentsList__message">No comments...</p>;
	}
	return (
		<ul className="CommentsList">
			{comments.map(({ author, comment, timestamp }: IComment, index) => (
				<li className="CommentsList__item" key={`comment-${index}`}>
					<Comment author={author} comment={comment} timestamp={timestamp} />
				</li>
			))}
		</ul>
	);
};

export default CommentsList;
