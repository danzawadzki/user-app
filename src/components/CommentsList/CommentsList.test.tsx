import React from 'react';
import { shallow } from 'enzyme';
import CommentsList from './CommentsList';

describe('CommentsList component: ', () => {
	//Shallow rendered component
	const user = {
		author: {
			name: 'Random User',
			avatar: {
				small: 'https://randomuser.me/api/portraits/thumb/men/65.jpg',
				regular: 'https://randomuser.me/api/portraits/med/men/65.jpg',
				large: 'https://randomuser.me/api/portraits/men/65.jpg'
			}
		},
		comment:
			'Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula.',
		timestamp: '10/10/2018'
	};

	const comments = [user, user, user, user, user];
	const component = shallow(<CommentsList comments={comments} />);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should have length of comments array', () => {
		expect(component.find('li')).toHaveLength(comments.length);
	});

	it('should render message "No comments...", if no comments were provided', () => {
		const componentEmpty = shallow(<CommentsList />);

		expect(componentEmpty.html()).toContain('No comments...');
	});
});
