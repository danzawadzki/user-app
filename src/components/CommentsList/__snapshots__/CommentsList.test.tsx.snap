// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsList component:  should render match snapshot 1`] = `
<ul
  className="CommentsList"
>
  <li
    className="CommentsList__item"
    key="comment-0"
  >
    <Comment
      author={
        Object {
          "avatar": Object {
            "large": "https://randomuser.me/api/portraits/men/65.jpg",
            "regular": "https://randomuser.me/api/portraits/med/men/65.jpg",
            "small": "https://randomuser.me/api/portraits/thumb/men/65.jpg",
          },
          "name": "Random User",
        }
      }
      comment="Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula."
      timestamp="10/10/2018"
    />
  </li>
  <li
    className="CommentsList__item"
    key="comment-1"
  >
    <Comment
      author={
        Object {
          "avatar": Object {
            "large": "https://randomuser.me/api/portraits/men/65.jpg",
            "regular": "https://randomuser.me/api/portraits/med/men/65.jpg",
            "small": "https://randomuser.me/api/portraits/thumb/men/65.jpg",
          },
          "name": "Random User",
        }
      }
      comment="Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula."
      timestamp="10/10/2018"
    />
  </li>
  <li
    className="CommentsList__item"
    key="comment-2"
  >
    <Comment
      author={
        Object {
          "avatar": Object {
            "large": "https://randomuser.me/api/portraits/men/65.jpg",
            "regular": "https://randomuser.me/api/portraits/med/men/65.jpg",
            "small": "https://randomuser.me/api/portraits/thumb/men/65.jpg",
          },
          "name": "Random User",
        }
      }
      comment="Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula."
      timestamp="10/10/2018"
    />
  </li>
  <li
    className="CommentsList__item"
    key="comment-3"
  >
    <Comment
      author={
        Object {
          "avatar": Object {
            "large": "https://randomuser.me/api/portraits/men/65.jpg",
            "regular": "https://randomuser.me/api/portraits/med/men/65.jpg",
            "small": "https://randomuser.me/api/portraits/thumb/men/65.jpg",
          },
          "name": "Random User",
        }
      }
      comment="Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula."
      timestamp="10/10/2018"
    />
  </li>
  <li
    className="CommentsList__item"
    key="comment-4"
  >
    <Comment
      author={
        Object {
          "avatar": Object {
            "large": "https://randomuser.me/api/portraits/men/65.jpg",
            "regular": "https://randomuser.me/api/portraits/med/men/65.jpg",
            "small": "https://randomuser.me/api/portraits/thumb/men/65.jpg",
          },
          "name": "Random User",
        }
      }
      comment="Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula."
      timestamp="10/10/2018"
    />
  </li>
</ul>
`;
